pkg/perf/healthcheck/healthcheck.go

```diff
@@ -0,0 +1,97 @@
+package healthcheck
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"github.com/rs/zerolog/log"
+	"github.com/threefoldtech/zos/pkg/app"
+	"github.com/threefoldtech/zos/pkg/perf"
+	"github.com/threefoldtech/zos/pkg/stubs"
+)
+
+const (
+	id          = "healthcheck"
+	schedule    = "0 */20 * * * *"
+	description = "health check task runs multiple checks to ensure the node is in a usable state and set flags for the power daemon to stop reporting uptime if it is not usable"
+)
+
+// NewTask returns a new health check task.
+func NewTask() perf.Task {
+	checks := []checkFunc{
+		cacheCheck,
+	}
+	return &healthcheckTask{
+		checks: checks,
+		errors: make(map[string][]string),
+	}
+}
+
+type checkFunc func(context.Context) (string, error)
+
+type healthcheckTask struct {
+	checks []checkFunc
+	errors map[string][]string
+}
+
+var _ perf.Task = (*healthcheckTask)(nil)
+
+// ID returns task ID.
+func (h *healthcheckTask) ID() string {
+	return id
+}
+
+// Cron returns task cron schedule.
+func (h *healthcheckTask) Cron() string {
+	return schedule
+}
+
+// Description returns task description.
+func (h *healthcheckTask) Description() string {
+	return description
+}
+
+// Run executes the health checks.
+func (h *healthcheckTask) Run(ctx context.Context) (interface{}, error) {
+	log.Debug().Msg("starting health check task")
+	for k := range h.errors {
+		// reset errors on each run
+		h.errors[k] = make([]string, 0)
```

Comment:
doing 
```suggestion
		h.errors[k] = nil
```
is faster and has the same effect in your case.

Reply:
remeber that a `nil` array is a valid `empty` array. a call to `a = append(a, v)` will initialize the array automatically if needed.